### PR TITLE
fix: trim a new line to avoid making the invalid autofix result

### LINKED ISSUE #1514: 📝 `noUselessFragments` autofix causes syntax error with strings
### Environment information

```bash
CLI:
  Version:                      1.5.1
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/4.0.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Configuration

```JSON
{
  "formatter": {
    "enabled": true,
    "indentStyle": "space",
    "indentWidth": 2,
    "lineWidth": 100
  },
  "javascript": {
    "formatter": {
      "enabled": true,
      "quoteStyle": "single",
      "arrowParentheses": "asNeeded"
    }
  }
}
```

### Demonstration 

(just a demonstration, recreating the situation is not accurate)

```jsx
// state before autofix
const SomeComponent = () => {
  return (
    <React.Fragment>
      Your Data Dispute Request was sent to your current Employer
    </React.Fragment>
  );
}

// state after autofix
const SomeComponent = () => {
  return (
    "
      Your Data Dispute Request was sent to your current Employer
    "
  );
}
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by togami2864**
## Summary
close #1514 
Fixes https://github.com/biomejs/biome/issues/1144

## Test Plan
add snaps for cases

### CODE REVIEW

File: crates/biome_js_analyze/src/semantic_analyzers/complexity/no_useless_fragments.rs [hunk 1]
```diff
@@ -257,7 +257,7 @@ impl Rule for NoUselessFragments {
                         jsx_tag_expression(AnyJsxTag::JsxSelfClosingElement(node)).into_syntax(),
                     ),
                     AnyJsxChild::JsxText(text) => {
-                        let new_value = format!("\"{}\"", text.value_token().ok()?);
+                        let new_value = format!("\"{}\"", text.value_token().ok()?.text().trim());
```
Comment (Conaclos) [inline crates/biome_js_analyze/src/semantic_analyzers/complexity/no_useless_fragments.rs:260]:
```suggestion
                        let new_value = format!("\"{}\"", text.value_token().ok()?.text_trimmed());
```

Comment (togami2864) [reply crates/biome_js_analyze/src/semantic_analyzers/complexity/no_useless_fragments.rs]: I tried it, but `_trimmed` method didn't trim '\n' in this case.
```
e.g.
text.value_token().ok()?.text_trimmed()
text.value_token().ok()?.token_text_trimmed()
```

Comment (Conaclos) [reply crates/biome_js_analyze/src/semantic_analyzers/complexity/no_useless_fragments.rs]: Oh I see. The newline is part of the string.

**Review by Conaclos — APPROVED** [review]

**Comment by Conaclos** [comment]
@togami2864
Could you add a changelog entry under `linter / Bug fixes` and then run `just gen-web`?

**Review by ematipico — APPROVED** [review]

**Comment by Conaclos** [comment]
Thanks!

**Pull request merged**